### DISCUSSION TITLE
query custom post types by custom taxonomy

### DISCUSSION THREAD

**Discussion by popeating** [wp-graphql/wp-graphql#2289] [Advanced Custom Fields]
I have creted a custom post type with two custom taxonomies, and had them exposed to graphql.
They show correctly on the main tree, i can query the custom post type, the custom taxonomies and the custom post type contains the  associated custom taxonomy.
But it looks i am missing a way to query the custom post type using the custom taxonomy, in the "where"  possibilities there are not the custom taxonomy (as there are categories and tags in post)
Is there a way to add this way to filter?
i know i can query the custom taxonomy and show the associated custom post, but i wish i could keep the query 'straight' as i do for normal posts

Thank you

**Comment by Mario-Boss** [comment]
I wonder if you found any good solution to this problem..? I tried doing a similar query to find a custom post that matches the selected custom taxonomy.. no luck so far..

**Reply by Ulisses97** [reply]
I'm also having this problem. I need to filter some "posts" that have a specific value from a custom taxonomy

**Reply by lehn-etracker** [reply]
Same here!

**Reply by phkoon** [reply]
Update: **It's now a native feature of Elementor, since version 3.21**

**Comment by predaytor** [comment]
https://github.com/wp-graphql/wp-graphql-tax-query
For example, filter post type `Product` with a custom taxonomy `product-category` assigned to it:
```graphql
query Products($category: [String], $operator: TaxQueryOperator = IN) {
    products(
        where: {
            orderby: { field: DATE, order: ASC }
            taxQuery: { taxArray: { taxonomy: PRODUCTCATEGORY, field: SLUG, operator: $operator, terms: $category } }
        }
    ) {
        edges {
            node {
                id
                uri
                title
                date
                categories: productCategories {
                    nodes {
                        id
                        name
                    }
                }
                featuredImage {
                    node {
                        id
                    }
                }
            }
        }
    }
}
```
If category is empty (similar to `categoryName` when querying `posts`) to display regardless of category, change `$operator` to `EXISTS`.

**Reply by Kieran-Hyphen** [reply]
Thanks for the help!

**Reply by brunom63** [reply]
My good man. You have no idea how much this helped me.  Thanks!!

**Comment by phkoon** [comment]
**Update:**
### It's now a native feature of Elementor, since version 3.21
